x-pack/plugins/lens/public/drag_drop/drag_drop.tsx

```diff
@@ -175,350 +336,383 @@ const DragDropInner = React.memo(function DragDropInner(
+    if (!activeDropTargetMatches) {
+      setActiveDropTarget(value);
     }
   };
 
   const dragLeave = () => {
-    setState({ ...state, isActive: false, dragEnterClassNames: '' });
+    setActiveDropTarget(undefined);
   };
 
-  const drop = (e: DroppableEvent) => {
+  const drop = (e: DroppableEvent | React.KeyboardEvent<HTMLButtonElement>) => {
     e.preventDefault();
     e.stopPropagation();
 
-    setState({ ...state, isActive: false, dragEnterClassNames: '' });
-    setDragging(undefined);
-
-    if (onDrop && droppable) {
+    if (onDrop && droppable && dragging) {
       trackUiEvent('drop_total');
-      onDrop(dragging);
+      onDrop(dragging, value);
     }
+    setActiveDropTarget(undefined);
+    setDragging(undefined);
+    setKeyboardMode(false);
   };
+  return (
+    <>
+      {React.cloneElement(children, {
+        'data-test-subj': dataTestSubj || 'lnsDragDrop',
+        className: classNames(children.props.className, classes, className),
+        onDragOver: dragOver,
+        onDragLeave: dragLeave,
+        onDrop: drop,
+        draggable,
+      })}
+    </>
+  );
+});
 
-  const isReorderDragging = !!(dragging && itemsInGroup?.includes(dragging.id));
+const ReorderableDrag = memo(function ReorderableDrag(
+  props: DragInnerProps & { reorderableGroup: DragDropIdentifier[]; dragging?: DragDropIdentifier }
+) {
+  const {
+    reorderState: { isReorderOn, reorderedItems, direction },
+    setReorderState,
+  } = useContext(ReorderContext);
 
-  if (
-    draggable &&
-    itemsInGroup?.length &&
-    itemsInGroup.length > 1 &&
-    value?.id &&
-    dropTo &&
-    (!dragging || isReorderDragging)
-  ) {
-    const { label } = props as DraggableProps;
-    return (
-      <ReorderableDragDrop
-        dropTo={dropTo}
-        label={label}
-        className={className}
-        dataTestSubj={props['data-test-subj'] || 'lnsDragDrop'}
-        draggingProps={{
-          className: classNames(children.props.className, classes),
-          draggable,
-          onDragEnd: dragEnd,
-          onDragStart: dragStart,
-          isReorderDragging,
-        }}
-        dropProps={{
-          onDrop: drop,
-          onDragOver: dragOver,
-          onDragLeave: dragLeave,
-          dragging,
-          droppable,
-          itemsInGroup,
-          id: value.id,
-          isActive: state.isActive,
-        }}
-      >
-        {children}
-      </ReorderableDragDrop>
-    );
-  }
-  return React.cloneElement(children, {
-    'data-test-subj': props['data-test-subj'] || 'lnsDragDrop',
-    className: classNames(children.props.className, classes, className),
-    onDragOver: dragOver,
-    onDragLeave: dragLeave,
-    onDrop: drop,
-    draggable,
-    onDragEnd: dragEnd,
-    onDragStart: dragStart,
-  });
-});
+  const {
+    value,
+    setActiveDropTarget,
+    label = '',
+    keyboardMode,
+    isDragging,
+    activeDropTarget,
+    reorderableGroup,
+    onDrop,
+    setA11yMessage,
+  } = props;
```

Comment:
hmmm, is this comment is on the right line? I don't see any spread operator close 🤔  Could you please take a look?

Reply:
Yes I meant destructuring! Like I said this is more of a stylistic comment, because sometimes we destructure _too many properties_. For example, I would want to avoid this style of code:

```
const { a, b, c, d, e, f, g } = props;

if (a) {
  ...
}

return anotherFunction({ a, b, c, d });
```

In that case, do we really need to destructure b, c, and d? What about:

```
const { a, e, f, g, ...functionProps } = props;

if (a) {
   ...
}

return anotherFunction({ a, ...functionProps });
```

Reply:
Ahhh, the famous apropcalypse... Yeah, I also notice that and we might want to address it at some point but I feel like in this case the destructured props are thrown in so many different places that it wouldn't make sense. I'll take a deeper look.